eng/common/pipelines/templates/stages/archetype-typespec-emitter.yml

```diff
@@ -0,0 +1,452 @@
+parameters:
+# Path to the emitter package. This is used as the base path for script invocations.
+- name: EmitterPackagePath
+  type: string
+
+# Pool to use for the pipeline stages
+- name: Pool
+  type: object
+
+# Whether to build alpha versions of the packages. This is passed as a flag to the build script.
+- name: BuildPrereleaseVersion
+  type: boolean
+  default: true
+
+# Whether to use the `next` version of TypeSpec. This is passed as a flag to the initialize script.
+- name: UseTypeSpecNext
+  type: boolean
+  default: false
+
+# Custom steps to run after the repository is cloned but before other job steps.
+- name: InitializationSteps
+  type: stepList
+  default: []
+
+# Indicates the build matrix to use for post-build autorest validation
+- name: TestMatrix
+  type: object
+  default: {}
+
+# Whether to run the publish and regeneration stages. If false, only the build stage will run.
+- name: ShouldPublish
+  type: boolean
+  default: false
+
+# List of packages to publish. Each package is an object with the following properties:
+#   name: The name of the package. This is used to determine the name of the file to publish.
+#   type: The type of package. Currently supported values are 'npm' and 'nuget'.
+#   file: The path to the file to publish. This is relative to the packages directory in the build artifacts directory.
+- name: Packages
+  type: object
+  default: []
+
+# Whether to publish to the internal feed.
+- name: PublishInternal
+  type: boolean
+  default: true
+
+# Indicates if the Publish stage should depend on the Test stage
+- name: PublishDependsOnTest
+  type: boolean
+  default: false
+
+# Whether to publish to the internal feed.
+- name: ShouldRegenrate
+  type: boolean
+  default: false
+
+# Number of jobs to generate. This is the maximum number of jobs that will be generated. The actual number of jobs will be reduced if it would result in fewer than MinimumPerJob packages per job.
+- name: RegenerationJobCount
+  type: number
+  default: 10
+
+# Minimum number of packages to generate per job.
+- name: MinimumPerJob
+  type: number
+  default: 10
+
+# Indicates if regenration matrix should only contain folders with typespec files
+- name: OnlyGenerateTypespec
+  type: boolean
+  default: false
+
+stages:
+
+# Build stage
+# Responsible for building the autorest generator and typespec emitter packages
+# Produces the artifact `build_artifacts` which contains the following:
+#   package-versions.json: Contains a map of package name to version for the packages that were built
+#   overrides.json: Contains npm package version overrides for the emitter and generator
+#   packages/: Contains the packages to publish
+#   lock-files/: Contains package.json and package-lock.json files for use in the test stage to ensure tests are run against the correct dependency versions
+- stage: Build
+  pool: ${{ parameters.Pool }}
```

Comment:
Does this work in a 1es context? Since usually we have had to fully hardcode the pool no matter what.

Reply:
I think we'll need to see.  I would expect it to be resolved before passing it to the underlying template as it's a hard parameter in

Reply:
Or...  I did work in my test runs